operations/compare-helm-with-jsonnet/plugins/resolve-config/main.go

```diff
@@ -0,0 +1,349 @@
+}
+
+// DeploymentOrStatefulSet is a subset from k8s.io/api/apps/v1.Deployment and k8s.io/api/apps/v1.StatefulSet
+// copied here in order to avoid a dependency on the k8s.io/api/apps/v1 package
+type DeploymentOrStatefulSet struct {
+	Spec struct {
+		Template struct {
+			Spec PodSpec `yaml:"spec"`
+		} `yaml:"template"`
+	} `yaml:"spec"`
+}
+
+func (c *ConfigExtractor) ExtractPodSpec(obj *yaml.RNode) (PodSpec, bool, error) {
+	switch obj.GetKind() {
+	case "Deployment", "StatefulSet":
+		var dep = new(DeploymentOrStatefulSet)
+		err := decodeTypedObject(obj, &dep)
+		if err != nil {
+			return PodSpec{}, false, err
+		}
+		return dep.Spec.Template.Spec, true, nil
+	}
+
+	return PodSpec{}, false, nil
+}
+
+func (c *ConfigExtractor) ResolveConfigs() ([]*yaml.RNode, error) {
+	results := make(chan *yaml.RNode, len(c.allItems))
+
+	defaultObj, _, err := c.resolveArgsAndConfigFile(nil, "")
+	if err != nil {
+		return nil, err
+	}
+
+	err = concurrency.ForEachJob(context.Background(), len(c.allItems), runtime.NumCPU(), func(ctx context.Context, idx int) error {
+		pod, ok, err := c.ExtractPodSpec(c.allItems[idx])
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return nil
+		}
+
+		for _, container := range pod.Containers {
+			if !strings.HasPrefix(container.Image, MimirImage) {
+				continue
+			}
+
+			var configFileText string
+			mountPath, args := findConfigFileArg(container.Args)
+			if mountPath != "" {
+				for _, vm := range container.VolumeMounts {
+					if !strings.HasPrefix(mountPath, vm.MountPath) {
+						continue
+					}
+
+					content, err := c.resolveConfigFileText(vm.Name, filepath.Base(mountPath), pod)
+					if err != nil {
+						return errors.Wrapf(err, "failed to resolve volume mount %s in pod", vm.Name)
+					}
+
+					configFileText = content
+				}
+			}
+
+			configObj, target, err := c.resolveArgsAndConfigFile(args, configFileText)
+			if err != nil {
+				return errors.Wrap(err, "failed to resolve config")
+			}
+
+			err = annotateDefaults(configObj, defaultObj)
+			if err != nil {
+				return errors.Wrap(err, "failed to annotate defaults")
+			}
+
+			resultNode := yaml.MustParse(`
+apiVersion: grafana.com/v1alpha1
+kind: MimirConfig
+metadata:
+  name: mimir-config
+  namespace: default
+`)
+			err = resultNode.SetName(target)
+			if err != nil {
+				return errors.Wrap(err, "failed to set name")
+			}
+			err = resultNode.SetMapField(configObj, "config")
+			if err != nil {
+				return errors.Wrap(err, "failed to set config")
+			}
+
+			results <- resultNode
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	close(results)
+
+	rawConfigs := []*yaml.RNode{}
+	for r := range results {
+		rawConfigs = append(rawConfigs, r)
+	}
+
+	return rawConfigs, nil
+}
+
+func (c *ConfigExtractor) resolveArgsAndConfigFile(args []string, configFileText string) (*yaml.RNode, string, error) {
+	var mimirConfig = &mimir.Config{}
+
+	flagSet := flag.NewFlagSet("mimir", flag.ContinueOnError)
```

Comment:
I'm not sure about `ContinueOnError`. Shouldn't all flags successfully marshal into a mimir config?